Unittest.py

```diff
@@ -68,6 +80,164 @@ def load_spoiler(json_file):
+                    spoiler_value = spoiler['locations'][location]
+                    if isinstance(item_value, dict):
+                        item_list = item_value['item']
+                    else:
+                        item_list = item_value
+                    if isinstance(spoiler_value, dict):
+                        self.assertIn(spoiler_value['item'], item_list)
+                    else:
+                        self.assertIn(spoiler_value, item_list)
+
+    def test_num_limited_items(self):
+        filenames = [
+            "plando-num-bottles-fountain-closed-bad",
+            "plando-num-bottles-fountain-open-bad",
+            "plando-num-adult-trade-item-bad",
+            "plando-num-weird-egg-item-bad"
+        ]
+        for filename in filenames:
+            with self.subTest(filename):
+                self.assertRaises(RuntimeError, generate_with_plandomizer, filename)
+
+    def test_excess_starting_items(self):
+        distribution_file, spoiler = generate_with_plandomizer("plando-excess-starting-items")
+        excess_item = list(distribution_file['starting_items'])[0]
+        for location, item in spoiler['locations'].items():
+            if isinstance(item, dict):
+                test_item = spoiler['locations'][location]['item']
+            else:
+                test_item = spoiler['locations'][location]
+            self.assertNotIn(excess_item, test_item)
+        self.assertNotIn(excess_item, spoiler['item_pool'])
+
+    def test_ammo_max_out_of_bounds_use_last_list_element(self):
+        # This issue only appeared while patching
+        filename = "plando-ammo-max-out-of-bounds"
+        settings = Settings({
+            'enable_distribution_file': True,
+            'distribution_file': os.path.join(test_dir, filename + '.json'),
+            'compress_rom': "Patch",
+            'count': 1,
+            'create_spoiler': True,
+            'create_cosmetics_log': False,
+            'output_file': os.path.join(test_dir, 'Output', filename),
+            'seed': 'TESTTESTTEST'
+        })
+        main(settings)
+
+    def test_ice_trap_has_model(self):
+        filenames = [
+            "plando-item-pool-matches-items-placed-after-starting-items-replaced",
+            "plando-new-placed-ice-traps",
+            "plando-placed-and-added-ice-traps"
+        ]
+        for filename in filenames:
+            with self.subTest(filename):
+                distribution_file, spoiler = generate_with_plandomizer(filename)
+                locations_with_previews = location_groups['CanSee']
+                for location in locations_with_previews:
+                    if location in spoiler['locations']:
+                        item = spoiler['locations'][location]
+                        if isinstance(item, dict) and item['item'] == "Ice Trap":
+                            self.assertIn("model", item)
+                        else:
+                            self.assertNotIn("Ice Trap", item)
```

Comment:
item could still be a dict here. Not sure if that's okay.

Reply:
This shouldn't be an issue for this particular case, but I can separate the and part of the if into another if nested inside of the first to help clarify only dicts with items of Ice Trap matter.

Reply:
Yeah, I think I like that better.